test/extensions/upstreams/tcp/generic/config_test.cc

```diff
@@ -75,37 +86,43 @@ TEST_F(TcpConnPoolTest, TestTunnelingNotDisabledIfFilterStateHasFalseValue) {
       std::make_shared<StreamInfo::BoolAccessorImpl>(false),
       StreamInfo::FilterState::StateType::Mutable, StreamInfo::FilterState::LifeSpan::Connection);
 
-  EXPECT_CALL(thread_local_cluster_, httpConnPool(_, _, _)).WillOnce(Return(absl::nullopt));
+  if (!GetParam()) {
```

Comment:
one final optional nit - if you can have a class helper function which makes it clear what's going on here (useUpstreamFilters()) or something that'd be a bit more readable.